### PR TITLE
Add `not_found` tag for shortcodes

### PR DESCRIPTION
**Pull request by JoryHogeveen**
## Description
Fixes: #5580 

## ChangeLog

* Enhancement: Add `not_found` tag to the Pods shortcode to return default string if no output is empty.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

File: includes/general.php [hunk 1]
```diff
@@ -1083,7 +1084,13 @@ function pods_shortcode_run( $tags, $content = null ) {
 		echo $pod->pagination( $tags['pagination_label'] );
 	}
 
-	echo $pod->template( $tags['template'], $content );
+	$content = $pod->template( $tags['template'], $content );
```
Comment (sc0ttkclark) [inline includes/general.php]:
```suggestion
	$content = $pod->template( $tags['template'], $content );
	$content = trim( $content );
```

File: includes/general.php [hunk 2]
```diff
@@ -1083,7 +1084,14 @@ function pods_shortcode_run( $tags, $content = null ) {
...
+	$content = trim( $content );
+
+	if ( empty( $content ) && ! empty( $tags['not_found'] ) ) {
+		echo $pod->do_magic_tags( $tags['not_found'] );
+	} else {
+		echo $content;
+	}
```
Comment (sc0ttkclark) [inline includes/general.php]:
```suggestion
	if ( empty( $content ) && ! empty( $tags['not_found'] ) ) {
		$content = $pod->do_magic_tags( $tags['not_found'] );
	}

	echo $content;
```

File: includes/general.php [hunk 3]
```diff
@@ -1090,7 +1091,14 @@ function pods_shortcode_run( $tags, $content = null ) {
...
 
-	echo $pod->template( $tags['template'], $content );
+	$content = $pod->template( $tags['template'], $content );
+	$content = trim( $content );
+
+	if ( empty( $content ) && ! empty( $tags['not_found'] ) ) {
+		$content = $pod->do_magic_tags( $tags['not_found'] );
+	}
+
+	echo $content;
```
Comment (tr1b0t) [inline includes/general.php]: * :warning: Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$content'
* :no_entry_sign: All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found '$content'.

<sub>hash: 1100ff42becac100619afdca4cbf44de</sub>

Comment (sc0ttkclark) [inline includes/general.php]:
```suggestion
	// phpcs:ignore
	echo $content;
```

**Pull request merged**